tutorial/source/dirichlet_process_mixture.ipynb

```diff
@@ -0,0 +1,799 @@
ylpziMdWp9BOhFYqyNIJ6TnATeS2tXv7GW8rc0S++ZY5+fYTi/E9lXSD85S\nUtv1tV2scjPSCfUXSU1Ld1KoybfjKtJJxOI8g0j7ZT5p3+xKavZB0p65Oao9K0g/3neRThzPyH+/\nmMv6KPBD0kn82aQT5rUaTDpH8nfShQPzgXMK068gJfzFwAfI519yBeNg4ORcln8mnRx/IS93LnBE\nbmYrro+8fACHALcD15E+1xmkdvgpebbDgG1I++464JSImJqnXQo8Rmr++TXpCK4muZn0emBuju8d\nwP7AY/n7cBZwWES8Wus6m5EKR7xmVhGSrgDmRMSERsdi9ecav5lZxTjxm5lVjJt6zMwqxjV+M7OK\naWgHVbUaMmRIjBo1qtFhmJkNKA8++ODiiFilB4ABkfhHjRrFtGnTup7RzMzeJKntHdiAm3rMzCrH\nid/MrGKc+M3MKsaJ38ysYpz4zcwqxonfzKxiSkv8kkbkJwk9KmmWpBPz+An5YRfT86vyD0UwM6un\nMq/jfx34ZkQ8lPvGflDSHXnajyPirBK3bWZmHSgt8ecHOszPw0slPUb3H7tnZmZ9rC537koaBewA\n3E96APcJkr5AegrPNwsPeCguMw4YBzBy5MjSYhs1fnLXM3Vg7sRanz1tZtZ/lH5yNz/H9XrgpPxA\n6vNJTxwaQzoiOLu95SJiUkS0RETL0KGrdDVhZmY9VGril7QGKelfGRE3AETEgohYmZ9/eSGwU5kx\nmJnZ25V5VY9Iz/t8LCLOKYwfXpjt08DMsmIwM7NVldnGvxtwFDBD0vQ87hTSQ5rHkJ50Pxf4Sokx\nmJlZG2Ve1XMvoHYm3VrWNs3MrGu+c9fMrGKc+M3MKsaJ38ysYpz4zcwqxonfzKxinPjNzCrGid/M\nrGKc+M3MKsaJ38ysYpz4zcwqxonfzKxinPjNzCrGid/MrGKc+M3MKsaJ38ysYpz4zcwqxonfzKxi\nnPjNzCrGid/MrGKc+M3MKsaJ38ysYpz4zcwqxonfzKxinPjNzCrGid/MrGKc+M3MKsaJ38ysYlZv\ndAAD2ajxkzudPnfiAXWKxMysdq7xm5lVjBO/mVnFOPGbmVWME7+ZWcWUlvgljZB0l6RHJc2SdGIe\nv4mkOyQ9kf9uXFYMZma2qjJr/K8D34yI0cDOwPGSRgPjgSkRsQ0wJb83M7M6KS3xR8T8iHgoDy8F\nHgM2B8YCl+XZLgM+VVYMZma2qrq08UsaBewA3A8Mi4j5edJzwLAOlhknaZqkaYsWLapHmGZmlVB6\n4pe0PnA9cFJEvFicFhEBRHvLRcSkiGiJiJahQ4eWHaaZWWWUmvglrUFK+ldGxA159AJJw/P04cDC\nMmMwM7O3K/OqHgEXAY9FxDmFSbcAR+fho4Gby4rBzMxWVWZfPbsBRwEzJE3P404BJgLXSjoOeAr4\nXIkxmJlZG6Ul/oi4F1AHkz9e1nbNzKxzvnPXzKxinPjNzCrGid/MrGKc+M3MKsaJ38ysYpz4zcwq\nxonfzKxinPjNzCrGid/MrGKc+M3MKsaJ38ysYpz4zcwqpsvEL+l0SRtKWl3SbyQtkHRkPYIzM7O+\nV0uNf7/85KwDgXnA+4CTS43KzMxKU0viXyP/3R/4n4h4gQ4el2hmZv1fLf3xT5Y0E1gJHC9pCLCi\n3LDMzKwsndb4Ja1GembuXsCHIuI1YDlwSB1iMzOzEnSa+CPiDeBnEbEwIl7P416KiGfrEp2ZmfW5\nWtr475I0tvRIzMysLmpp4z8GOFHSCuAV0nN0IyI2KTMwMzMrRy2Jf0jpUZiZWd10mfgjYqWkjYCt\ngbULk35fWlRmZlaaLhO/pOOAbwCbAzOADwP3AXuWGpmZmZWilpO7JwEtwNyI+CjwIeBvpUZlZmal\nqSXxL4+IVwAkrRkRs4D3lhuWmZmVpZaTu/MlDQZ+CfxG0vPAM+WGZWZmZanl5O7BefC7kj4ObARM\nLjUqMzMrTS01fiTtDLwnIi6XtCkwDPhrqZH1kVHj++dvVFdxzZ14wIBbd2/Wa2b1U8tVPd8BdiNd\nznk56ZLOq4Ddyw3NzMzKUMvJ3UNJXTIvA8j99GxYZlBmZlaeWhL/iogIch/8ktYtNyQzMytTLYn/\nBknnARtJOha4Hbi43LDMzKwsXSb+iDgD+BVwC/BB4N8j4tyulpN0saSF+SEureMmSHpW0vT82r83\nwZuZWfd1eHJX0vnAyRHxYkTcBtzWzXVfCvwn6YRw0Y8j4qxursvMzPpIZzX+J4EHJR3ZkxVHxD3A\n8z2KyszMStNh4o+IM0kdsY2VNEXSoZIOaX31YpsnSHokNwVt3Iv1mJlZD3T16MVnSXfpvgc4qPA6\nsIfbO590P8AYYD5wdkczShonaZqkaYsWLerh5szMrK3O2vi3JSXqecBOETG/txuLiAWF9V9IOmnc\n0byTgEkALS0t0dttm5lZ0tmdu9cBJ0bE7X21MUnDCz8gnwZmdja/mZn1vc4S/5iIWNHTFUu6mnSO\nYIikZ4BTgT0ljSHdDDYX+EpP129mZj3TYeLvTdLPyx/RzuiLerNOMzPrvVru3DUzsybSYeKXNCX/\nPaN+4ZiZWdk6a+MfLmlX4GBJ1wAqToyIh0qNzMzMStFZ4v8e8F3gXcA5baYFsFdZQZkfeNJWmQ+X\nMauazk7uXgdcJ+m7EXFaHWMyM7MS1fLM3dMkHQzskUdNjYgOb7wyM7P+rcureiSdDpwIPJpfJ0r6\nYdmBmZlZOWp52PoBpJu53gCQdBnwR+CUMgMzM7Ny1Hod/+DC8EZlBGJmZvVRS43/dOCPku4iXdK5\nBzC+1KjMzKw0tZzcvVrSVODDedTJEfFcqVGZmVlpaqnxk3vUvKXkWKxOfI+AWbW5rx4zs4px4jcz\nq5hOE7+kQZL+VK9gzMysfF09c3clMFvSyDrFY2ZmJavl5O7GwCxJD
```

Comment:
Beside fixing log_posterior_predictive_eval computation, could you review this formula? I am not sure but probably we should move the sum over `t` inside the expectation, and remove one duplicated E_q...

Reply:
This formula is correct, these are independent expectations and thus they are able to factor as such. Similarly, expectations are linear so we can take the sum outside of the expectation.

Reply:
Hmm, is it better to use a formula aligned with the implementation (without having to know those facts)? But I don't have a strong opinion on this. LGTM!